### PR TITLE
Analytics change inner blobs to json

### PR DESCRIPTION
**Pull request by pedronveloso**
This PR addresses a few things pertaining to Analytics ingestion. Namely:
 - The inner blobs of the payloads, `action` and `grants` should be JSON encoded & escaped for backend processing
 - We want to use the `action` from the request payload instead of the abstraction `PayKitPaymentAction` for analytics. Thus  we had to change some analytics to the network layer
 - There was some double-reporting for analytics on Update Request calls

# Verification

Looked at entry JSON validation under the Real Time event ingestion console. Furthermore, data on snowflake looks ok now 👍🏻

### CODE REVIEW

File: core/src/main/java/app/cash/paykit/core/analytics/PayKitAnalyticsEventDispatcherImpl.kt [hunk 1]
```diff
@@ -115,38 +117,10 @@ internal class PayKitAnalyticsEventDispatcherImpl(
...
-          clientId,
-          action = stateToAnalyticsAction(CreatingCustomerRequest),
-          createActions = action.toString(),
-          createChannel = CHANNEL_IN_APP,
-          createRedirectUrl = action.redirectUri,
-          createReferenceId = null,
-        )
-      }
-    }
+    val eventPayload = createOrUpdateAnalyticsPayload(paymentKitAction, apiAction, null)
```
Comment (pedronveloso) [inline core/src/main/java/app/cash/paykit/core/analytics/PayKitAnalyticsEventDispatcherImpl.kt:123]: Consolidated this and the below function under a single function `createOrUpdateAnalyticsPayload`

File: core/src/main/java/app/cash/paykit/core/models/analytics/payloads/AnalyticsCustomerRequestPayload.kt [hunk 2]
```diff
@@ -82,7 +82,7 @@ data class AnalyticsCustomerRequestPayload(
   val requestChannel: String? = null,
 
   // The ID of the Customer Request.
-  @Json(name = "mobile_cap_pk_customer_request_request_id")
+  @Json(name = "mobile_cap_pk_customer_request_customer_request_id")
```
Comment (pedronveloso) [inline core/src/main/java/app/cash/paykit/core/models/analytics/payloads/AnalyticsCustomerRequestPayload.kt:85]: This field was misreported - this is the correct name.

File: core/src/main/java/app/cash/paykit/core/CashAppPayKit.kt [hunk 3]
```diff
@@ -154,6 +154,7 @@ object CashAppPayKitFactory {
     )
     val analyticsEventDispatcher =
       buildPayKitAnalyticsEventDispatcher(clientId, networkManager, paykitAnalytics)
+    networkManager.analyticsEventDispatcher = analyticsEventDispatcher
```
Comment (cpetzel) [inline core/src/main/java/app/cash/paykit/core/CashAppPayKit.kt:157]: Why do we have to set this variable at runtime, and not during construction?

Comment (cpetzel) [reply core/src/main/java/app/cash/paykit/core/CashAppPayKit.kt:157]: Ah I see. SG.

**Review by cpetzel — APPROVED** [review]

**Pull request merged**